src/main/java/Entities/Implementations/AbstractRecipeItem.java

```diff
@@ -0,0 +1,46 @@
+package Entities.Implementations;
+
+import Entities.Ingredient;
+import EntityInterfaces.RecipeItem;
+
+public abstract class AbstractRecipeItem implements RecipeItem {
+    public Ingredient ingredient;
+    public float amount;
+    public boolean optional = false;
+
+    /** Implement a ItemImpl, giving the ingredient, and
+     * quantity of this ingredient.
+     * @param ingredient  The ingredient
+     * @param amount     The quantity of this ingredient
+     */
+    //public AbstractRecipeItem(Ingredient ingredient, float amount) {
+    //   this.ingredient = ingredient;
+    // this.amount = amount;
+    //}
+    /** Implement a ItemImpl, giving the ingredient,
+     * quantity of this ingredient, and if the ingredient is optional.
+     * @param ingredient  The ingredient
+     * @param amount     The quantity of this ingredient
+     * @param optional  Is the ingredient optional?
+     */
```

Comment:
Why is this constructor commented out?

These doc comments are poor. This is not about implementing a `ItemImpl`. The comment should not describe the obvious when possible. It should be plain english about the use case. A feasible thing to say would be this assumes the item is not optional since this is a choice made by the class when using that constructor.